src/Datadog.Trace/DuckTyping/DuckType.Methods.cs

```diff
@@ -22,18 +22,33 @@ private static List<MethodInfo> GetMethods(Type baseType)
                     continue;
                 }
 
-                List<MethodInfo> newMethods = imInterface.GetMethods()
-                    .Where(iMethod =>
+                foreach (MethodInfo intMethod in imInterface.GetMethods())
+                {
+                    if (intMethod.IsSpecialName)
+                    {
+                        continue;
+                    }
+
+                    string intMethodName = intMethod.ToString();
+                    bool found = false;
+                    foreach (MethodInfo currentMethod in selectedMethods)
```

Comment:
Suggestion: I think doing a LINQ `Any` might be a little more readable

```
if (selectedMethods.Any(method => method.ToString() == intMethodName))
{
    continue;
}
```

Reply:
What if I tell you I removed the LINQ on purpose? 😂 I wanted to remove the closure and also for a simple comparison. But no strong opinion about it.

Reply:
Haha that makes sense! I don't have a strong opinion either, it just took me a minute to realize what this loop was doing. Maybe just adding a small comment here or renaming the `bool found` to be a little more explanatory could help

Reply:
I did the renaming in 2719b3c